mobile-app/app/api/wallet/notifications.ts

```diff
@@ -0,0 +1,77 @@
+import * as Notifications from 'expo-notifications'
+import { Platform } from 'react-native'
+import { PermissionStatus } from 'expo-modules-core'
+
+export interface SendNotificationProps {
+  title: string
+  body: string
+  data?: { [key: string]: unknown }
+}
+
+/**
+ * Wallet push notification implementation light wallet
+ */
+class NotificationsService {
+  status: PermissionStatus | undefined
+
+   /**
+   * Register notifications permission for app.
+   */
+  async register (): Promise<void> {
+    Notifications.setNotificationHandler({
+      handleNotification: async () => ({
+        shouldShowAlert: true,
+        shouldPlaySound: false,
+        shouldSetBadge: false
+      })
+    })
+
+    if (Platform.OS !== 'web') {
+      const { status: existingStatus } = await Notifications.getPermissionsAsync()
+      let finalStatus = existingStatus
+      if (existingStatus !== PermissionStatus.GRANTED) {
+        const { status } = await Notifications.requestPermissionsAsync({
+          ios: {
+            allowAlert: true,
+            allowBadge: true,
+            allowAnnouncements: true
+          }
+        })
+        finalStatus = status
+      }
+
+      if (finalStatus !== PermissionStatus.GRANTED) {
+        return
+      }
+      this.status = finalStatus
+      if (Platform.OS === 'android') {
+        await Notifications.setNotificationChannelAsync('default', {
+          name: 'default',
+          importance: Notifications.AndroidImportance.MAX,
+          sound: null
+        })
+      }
+    }
+  }
+
+  /**
+   * Register notifications permission for app.
+   * @param {string} title
+   * @param {string} body
+   * @param {{ key: string]: unknown }} data
+   */
+  async send ({ title, body, data = {} }: SendNotificationProps): Promise<void> {
+    if (this.status === PermissionStatus.GRANTED) {
+      await Notifications.scheduleNotificationAsync({
+        content: {
+          title,
+          body,
+          data
+        },
+        trigger: null
+      })
+    }
+  }
+}
+
+export const WalletNotifications = new NotificationsService()
```

Comment:
it shouldn't be a class, should just be plain old functions?

Reply:
When our app loads, we first check the push notification permission using [register](https://github.com/DeFiCh/wallet/blob/a6ce2394b345a3df3234a0320e4d4940f5c02078/mobile-app/app/api/wallet/notifications.ts#L20) function and set the permission status to `status` variable, and before sending the notification we first check the same status is granted or not ([link](https://github.com/DeFiCh/wallet/blob/a6ce2394b345a3df3234a0320e4d4940f5c02078/mobile-app/app/api/wallet/notifications.ts#L64)). For storing the status, it's better we use class function, otherwise we have to store this in context or in redux store or in localStore.